creator-node/src/middlewares.js

```diff
@@ -389,6 +436,118 @@ async function getCreatorNodeEndpoints ({ serviceRegistry, logger, wallet, block
   return userReplicaSet
 }
 
+/**
+ * Retrieves user replica set spIDs from chain (POA.UserReplicaSetManager)
+ *
+ * Polls contract (via web3 provider) conditionally as follows:
+ *    - If `blockNumber` provided, polls contract until it has indexed that blockNumber (for up to 200 seconds)
+ *    - Else if `ensurePrimary` required, polls contract until it has indexed selfSpID as primary (for up to 60 seconds)
+ *      - Errors if retrieved primary spID does not match selfSpID
+ *    - If neither of above conditions are met, falls back to single contract query without polling
+ *
+ * @param {Object} serviceRegistry
+ * @param {Object} logger
+ * @param {string} wallet - wallet used to query discprov for user data
+ * @param {number} blockNumber - blocknumber of eth TX preceding CN call
+ * @param {string} selfSpID
+ * @param {boolean} ensurePrimary - determines if function should error if this CN is not primary
+ *
+ * @returns {Array} - array of strings of replica set
+ */
+async function getReplicaSetSpIDs ({ serviceRegistry, logger, wallet, userId, blockNumber, ensurePrimary, selfSpID }) {
+  const start = Date.now()
+  const logPrefix = `[getReplicaSetSpIDs] [userId = ${userId}]`
+  const { libs } = serviceRegistry
+
+  // returns Object of schema { primaryId: string, secondaryIds: int[] }
+  let replicaSet = null
+
+  /**
+   * If `blockNumber` provided, polls contract until it has indexed that blocknumber (for up to 200 seconds)
+   */
+  if (blockNumber) {
+    // In total, will try for 200 seconds.
+    const MaxRetries = 201
```

Comment:
I know this is not new to this PR, but MaxRetries to me reads as a class. Probably should be maxRetries

Reply:
ah right, i got my naming conventions wrong - seems like UPPER_CASE is the right move - done